### PR TITLE
Move to support fragments

### PR DESCRIPTION
**Pull request by quinnjn**
Android P will deprecate `android.app.Fragment`. This PR adds support for the support fragment `android.support.v4.app.Fragment`

This PR includes
* Updating to SDK 27
* BrowserSwitchSupportFragment which has the same functionality as BrowserSwitchFragment, just inside of a support fragment
* Deprecating BrowserSwitchFragment
* BrowserSwitch package-private class which holds shared logic between the two fragments

### CODE REVIEW

File: browser-switch/src/test/java/com/braintreepayments/browserswitch/test/TestFragmentActivity.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.braintreepayments.browserswitch.test;
+
+import android.support.v4.app.FragmentActivity;
+
+public class TestFragmentActivity extends FragmentActivity {
```
Comment (demerino) [inline browser-switch/src/test/java/com/braintreepayments/browserswitch/test/TestFragmentActivity.java]: Maybe helpful to add a comment here that this new class is used for tests and is intentionally left empty?

Comment (quinnjn) [reply browser-switch/src/test/java/com/braintreepayments/browserswitch/test/TestFragmentActivity.java]: Hmm we could, but it would only be visible for folks currently developing. 

It's in the `src/test/` directory and those files don't get published.

**Comment by quinnjn** [comment]
Closing in favor of #10
